src/mlpack/methods/time_series_forecast/exponential_smoothing/ses.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file methods/time_series/exponential_smoothing/ses.cpp
+ * @author Aditi Pandey
+ *
+ * Implementation of Single exponential smoothing.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core/util/log.hpp>
+#include "ses.hpp"
+#include <cmath>
+#include <numeric>
+
+using namespace mlpack;
+using namespace mlpack::ts;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+// Calculate errors
+
+double CalcMae(arma::rowvec& errors) {
+    double mae = 0.0;
+    for(arma::uword c=0; c < errors.n_elem; ++c){
+	mae += std::abs(errors(c));
+    }
+    return mae/errors.n_elem;
+}
+
+double CalcMse(arma::rowvec& errors) {
+    double mse = 0.0;
+    for(arma::uword c=0; c < errors.n_elem; ++c){
+	mse += pow(errors(c), 2);
+    }
+    return mse/errors.n_elem;
+}
+
+double CalcMape(arma::rowvec& data, arma::rowvec& errors) {
+    double mape = 0.0;
+    for(arma::uword c1=0, c2=0; c1 < errors.n_elem, c2 < data.n_elem; ++c1, ++c2){
+	    mape += std::abs(errors(c1) / data(c2));
+    }
+    return mape/errors.n_elem;
+}
+
+//Function declaration for the SES 
+
+SingleES::SingleES(arma::rowvec& data, const double& alpha) {
+    this->alpha = alpha;
```

Comment:
I think you can use initializer list here. This will eliminate the use of `this->`. If you are not familiar, then this might be helpful https://www.geeksforgeeks.org/when-do-we-use-initializer-list-in-c/.

Reply:
Oh awesome, got it 👍🏼